src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RollbacksSpec.groovy

```diff
@@ -112,4 +118,141 @@ and at least 1 path must remain safe"
         northbound.deleteLinkProps(northbound.getAllLinkProps())
         database.resetCosts()
     }
+
+    @Tidy
+    @Unroll
+    def "System tries to retry rule installation during #data.description if previous one is failed"(){
+        given: "Two active neighboring switches with two diverse paths at least"
+        def allPaths
+        def swPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            allPaths = it.paths
+            allPaths.unique(false) { a, b -> a.intersect(b) == [] ? 1 : 0 }.size() >= 2
+        }
+
+        List<PathNode> mainPath = allPaths.min { it.size() }
+        //find path with more than two switches
+        List<PathNode> protectedPath = allPaths.findAll { it != mainPath && it.size() != 2 }.min { it.size() }
+        assumeTrue("Unable to find a  required protectedPath", protectedPath as boolean)
+
+        and: "All alternative paths unavailable (bring ports down)"
+        def broughtDownPorts = []
+        def otherIsls = []
+        def involvedIsls = (pathHelper.getInvolvedIsls(mainPath) + pathHelper.getInvolvedIsls(protectedPath)).unique()
+        allPaths.findAll { it != mainPath && it != protectedPath }.each {
+            pathHelper.getInvolvedIsls(it).findAll { !(it in involvedIsls || it.reversed in involvedIsls) }.each {
+                otherIsls.add(it)
+            }
+        }
+        broughtDownPorts = otherIsls.unique { a, b -> a == b || a == b.reversed ? 0 : 1 }
```

Comment:
how 'isls' transform into 'ports' by just reducing the isls list? It is still 'brokenIsls' or 'broughtDownIsls' whatever but not 'ports'

Reply:
fixed